src/base/types/derivedmodels/SMomentModel.jl

```diff
@@ -0,0 +1,272 @@
+
+# Fields
+```
+reaction_ids::Vector{String}
+irrev_reaction_ids::Vector{String}
+metabolites::Vector{String}
+c::SparseVec
+S::SparseMat
+b::SparseVec
+xl::SparseVec
+xu::SparseVec
+C::SparseMat
+cl::Vector{Float64}
+cu::Vector{Float64}
+```
+"""
+mutable struct SMomentModel <: MetabolicModel
+    reaction_ids::Vector{String}
+    irrev_reaction_ids::Vector{String}
+    metabolites::Vector{String}
+    c::SparseVec
+    S::SparseMat
+    b::SparseVec
+    xl::SparseVec
+    xu::SparseVec
+end
+
+"""
+    stoichiometry(model::SMomentModel)
+
+Return stoichiometry matrix that includes enzymes as metabolites.
+"""
+stoichiometry(model::SMomentModel) = model.S
+
+"""
+    balance(model::SMomentModel)
+
+Return stoichiometric balance.
+"""
+balance(model::SMomentModel) = model.b
+
+"""
+    objective(model::SMomentModel)
+
+Return objective of `model`.
+"""
+objective(model::SMomentModel) = model.c
+
+"""
+    reactions(model::SMomentModel)
+
+Returns the reversible reactions in `model`. For 
+the irreversible reactions, use [`irreversible_reactions`][@ref].
+"""
+reactions(model::SMomentModel) = model.reaction_ids
+
+"""
+    n_reactions(model::SMomentModel)
+
+Returns the number of reactions in the model.
+"""
+n_reactions(model::SMomentModel) = length(model.reaction_ids)
+
+"""
+    irreversible_reactions(model::SMomentModel)
+
+Returns the irreversible reactions in `model`.
+"""
+irreversible_reactions(model::SMomentModel) = model.irrev_reaction_ids
+
+"""
+    metabolites(model::SMomentModel)
+
+Return the metabolites in `model`.
+"""
+metabolites(model::SMomentModel) = model.metabolites
+
+"""
+    n_metabolites(model::SMomentModel) = 
+
+Return the number of metabolites in `model`.
+"""
+n_metabolites(model::SMomentModel) = length(metabolites(model))
+
+"""
+    bounds(model::SMomentModel)
+
+Return variable bounds for `SMomentModel`.
+"""
+bounds(model::SMomentModel) = (model.xl, model.xu)
+
+"""
+    reaction_flux(model::MetabolicModel)
+
+Helper function to get fluxes from optimization problem.
+"""
+function reaction_flux(model::SMomentModel)
+    R = spzeros(n_reactions(model), length(model.irrev_reaction_ids) + 1)
+    for (i, rid) in enumerate(reactions(model))
+        for_idx = findfirst(
+            x -> x == rid * "§ARM§FOR" || x == rid * "§FOR",
+            model.irrev_reaction_ids,
+        )
+        rev_idx = findfirst(
+            x -> x == rid * "§ARM§REV" || x == rid * "§REV",
+            model.irrev_reaction_ids,
+        )
+        !isnothing(for_idx) && (R[i, for_idx] = 1.0)
+        !isnothing(rev_idx) && (R[i, rev_idx] = -1.0)
+    end
+    return R'
+end
+
+"""
+    SMomentModel(
+        model::StandardModel;
+        rid_isozymes = Dict{String, Vector{Isozyme}}(),
+    )
+
+Construct an `SMomentModel`.
+
+"""
+function SMomentModel(
+    model::StandardModel;
+    rid_isozymes = Dict{String,Vector{Isozyme}}(),
+    enzyme_capacity = 0.0,
+)
+
```

Comment:
same as with gecko -- materialize this as make_smoment_model and put to `analysis/smoment.jl`